src/pool/mod.rs

```diff
@@ -8,6 +8,41 @@ use crate::{
 
 use std::{fmt, marker::PhantomData};
 
+/// A lock-free concurrent object pool.
+///
+/// Slabs provide pre-allocated storage for many instances of a single type. But when working with
+/// heap allocated objects, the advantages of a Slab are lost as the memory allocated for the
+/// object is freed when the object is removed from the Slab. With a pool, we can instead re-use
+/// this memory for objects being added to the pool in the future, therefore reducing memory
+/// fragmentation and avoiding additional allocations.
+///
+/// This module implements a lock-free concurrent pool, indexed by `usize`s.
```

Comment:
How about
```suggestion
/// This type implements a lock-free concurrent pool, indexed by `usize`s.
```
since the `Pool` module is not going to be exposed publicly (we're reexporting the `Pool` type from the crate root).